### DISCUSSION TITLE
How to pass an edge template to MessageContract

### DISCUSSION THREAD

**Discussion by feniljariwala82** [adonisjs/core#3174] [Help] [answered]
**Purpose**: Pass edge template to MessageContract for email verification.
**Problem**: I am following https://docs.adonisjs.com/guides/mailer documentation. I am trying to do the same as shown in the video. But I am having the following issue! 

![Screenshot from 2021-09-15 22-36-53](https://user-images.githubusercontent.com/42708686/133478937-472ac5b9-3818-43cd-8514-c964daccfe7d.png)

Can anybody guide me here?

**Comment by thetutlage** [comment, marked as answer]
You can simply use the `htmlView` method and there is no need to render the template separately. 

```ts
message.subject().htmlView('emails/verify')
```

However, the real root cause of the issue you are facing is that the `View.render` now returns a Promise and it must be awaited before you can use it. So if you want to render the template separately, then you can do it as follows.

```ts
public async prepare (message) {
  const html = await View.render('emails/verify')

  message.html(html)
}
```

**Reply by feniljariwala82** [reply]
Thanks, @thetutlage, it worked. But now, I am having a different issue, when I send an email it throws this error.

**Error message**: "getaddrinfo EAI_AGAIN api.mailgun.net"

If you have any idea of solving this issue please let me know. It would be helpful. And thanks again for the help.
